quipucords/tests/integration/test_openshift_scan.py

```diff
@@ -0,0 +1,280 @@
+# Copyright (C) 2022  Red Hat, Inc.
+
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+
+"""Integration test for OpenShift scan."""
+
+import tarfile
+from io import BytesIO
+from logging import getLogger
+from time import sleep
+from unittest import mock
+
+import pytest
+
+from api.models import ScanTask
+from fingerprinter.constants import (
+    ENTITLEMENTS_KEY,
+    META_DATA_KEY,
+    PRODUCTS_KEY,
+    SOURCES_KEY,
+)
+from scanner.openshift.api import OpenShiftApi
+from scanner.openshift.entities import OCPDeployment, OCPProject
+from tests.utils.facts import RawFactComparator
+from utils import load_json_from_tarball
+
+logger = getLogger(__name__)
+
+
+@pytest.fixture
+def expected_facts():
+    """Set of expected facts on a OpenShift scan."""
+    return {}
+
+
+@pytest.fixture
+def expected_middleware_names():
+    """Middleware names."""
+    return {}
+
+
+@pytest.fixture
+def fingerprint_fact_map():
+    """Map fingerprint to raw fact name."""
+    return {
+        "name": "name",
+    }
+
+
+@pytest.fixture
+def expected_projects():
+    """Return a list with OCP projects."""
+    return [
+        OCPProject(
+            name="project name",
+            labels={"some": "label"},
+            deployments=[
+                OCPDeployment(
+                    name="deployment 1",
+                    labels={"n": "1"},
+                    container_images=["container-image-1:v1"],
+                    init_container_images=[],
+                ),
+                OCPDeployment(
+                    name="deployment 2",
+                    labels={"n": "2"},
+                    container_images=["container-image-1:v2"],
+                    init_container_images=["some-other-image:ver"],
+                ),
+            ],
+        ),
+    ]
+
+
+@pytest.fixture(autouse=True)
+def patched_openshift_client(mocker, expected_projects):
+    """Mock OpenShiftApi forcing it to return expected_projects."""
+    mocker.patch.object(OpenShiftApi, "can_connect", return_value=True)
+    mocker.patch.object(
+        OpenShiftApi,
+        "retrieve_projects",
+        return_value=expected_projects,
+    )
+    mocker.patch.object(
+        OpenShiftApi,
+        "retrieve_deployments",
+        return_value=expected_projects[0].deployments,
+    )
+
+
+@pytest.mark.integration
+@pytest.mark.django_db
+class TestOpenShiftScan:
+    """Smoke test network scan."""
```

Comment:
I this a remnant from the network scan test?

Reply:
yeah lol